src/test/java/pl/put/poznan/transformer/logic/serviceimpl/FindScenarioStepNamesNotStartingWithActorServiceImplTest.java

```diff
@@ -0,0 +1,271 @@
+        }
+
+        for(int i=0; i<correctSteps - 1; i++) {
+            steps.add(EXAMPLE_ACTOR + " " + EXAMPLE_USELESS_STATEMENT);
+        }
+
+        for(int i=0; i<stepsWithoutActorAtFirstPos; i++) {
+            steps.add(EXAMPLE_USELESS_STATEMENT);
+        }
+
+        return steps;
+    }
+
+    private ScenarioDTO prepareScenario(ArrayList<String> stepsName) {
+        ScenarioDTO scenarioDTO = new ScenarioDTO();
+        ArrayList<ScenarioStepDTO> scenarioSteps = new ArrayList<>();
+
+
+        for(String sName: stepsName) {
+
+            ScenarioStepDTO scenarioStepDTO = new ScenarioStepDTO();
+            scenarioStepDTO.setName(sName);
+            scenarioSteps.add(scenarioStepDTO);
+        }
+
+        ArrayList<String> actors = new ArrayList<>();
+        actors.add(EXAMPLE_ACTOR);
+
+        scenarioDTO.setActors(actors);
+        scenarioDTO.setSystemActor(EXAMPLE_SYSTEM_ACTOR);
+        scenarioDTO.setSteps(scenarioSteps);
+        return scenarioDTO;
+    }
+
+    private void addNestedSteps(ScenarioDTO scenarioDTO, int correctSteps, int incorrectSteps) {
+        ArrayList<String> stepsName = prepareStepsName(correctSteps, incorrectSteps);
+        ArrayList<ScenarioStepDTO> scenarioSteps = new ArrayList<>();
+
+        for(String sName: stepsName) {
+
+            ScenarioStepDTO scenarioStepDTO = new ScenarioStepDTO();
+            scenarioStepDTO.setName(sName);
+            scenarioSteps.add(scenarioStepDTO);
+        }
+
+        scenarioDTO.getSteps().get(0).setSteps(scenarioSteps);
+    }
+
+    private void addStepsWithKeywords(ScenarioStepDTO scenarioStepDTO) {
+        ArrayList<ScenarioStepDTO> steps = new ArrayList<>();
+        ScenarioStepDTO s;
+
+        s = new ScenarioStepDTO();
+        s.setName(EXAMPLE_IF_STEP_NAME);
+        steps.add(s);
+
+        s = new ScenarioStepDTO();
+        s.setName(EXAMPLE_ELSE_STEP_NAME);
+        steps.add(s);
+
+        s = new ScenarioStepDTO();
+        s.setName(EXAMPLE_FOR_EACH_STEP_NAME);
+        steps.add(s);
+
+        s = new ScenarioStepDTO();
+        s.setName(EXAMPLE__STEP_NAME_KEYWORD_IN_THE_MIDDLE);
+        steps.add(s);
+
+        scenarioStepDTO.setSteps(steps);
+    }
+
+    private ArrayList<String> retrieveIncorrectSteps(ScenarioDTO scenarioDTO) {
+        ArrayList<String> cSteps = new ArrayList<>();
+        Set<String> actors = new HashSet<String>();
+
+        actors.add(EXAMPLE_ACTOR);
+        actors.add(EXAMPLE_SYSTEM_ACTOR);
+
+
+        Function<ScenarioStepDTO, ArrayList<String>> getIncorrectSubSteps;
+        getIncorrectSubSteps = (ScenarioStepDTO subst) -> {
+
+            ArrayList<String> substNames = new ArrayList<>();
+
+            for(ScenarioStepDTO step: subst.getSteps()) {
+                if(!actors.stream().anyMatch(actor -> Pattern.compile("^" + actor + " ").matcher(step.getName()).find())) {
+                    cSteps.add(step.getName());
+                }
+            }
+
+            return substNames;
+        };
```

Comment:
This method creates ArrayList substNames, never assigns anything to it and returns it, you are using cSteps in here already instead I think anyway. But my concern is that you use some simmilar alorythm in your tests to test your initial service? Does it make sense to test service's method result with another result of another nearly identical method ?
Also it dosen't check if keyword is at the beggining or not as far as I'm concerned